### DISCUSSION TITLE
Getting the cloud list.

### DISCUSSION THREAD

**Discussion by JEMcats** [JEMcats/ubox_camera_api#10] [Q&A] [answered]
How would you format the curl command to get the cloud list?

**Comment by JEMcats** [comment, marked as answer]
to get the cloud list you can call uBox directly with:
```
curl --location 'https://portal.ubianet.com/api/user/cloud_list' \
--header 'x-ubia-auth-usertoken: Token Found In User Authentication' \
--header 'Content-Type: application/json' \
--data '{
        "summer_time": 1,
        "time_revised": true,
        "device_uid": [
            "UID of the device"
        ],
        "token": "Token Found In User Authentication",
        "time_diff": -18000,
        "page": 1,
        "timestamp": [
            1728619200,
            1728705664
        ]
    }'
```
but if you have the local server running (Recommend Method) with ```$ node server.js``` you can use:
```
curl --location 'http://localhost:8020/api/user/cloud_list' \
--header 'Content-Type: application/json' \
--data '{
        "summer_time": 1,
        "time_revised": true,
        "device_uid": [
            "UID of the device"
        ],
        "time_diff": -18000,
        "page": 1,
        "timestamp": [
            1728619200,
            1728705664
        ]
    }'
```
replace the values with valid information.

**Reply by ChristopherOleson** [reply]
When I run this I get error Invalid JSON

**Reply by JEMcats** [reply]
Thank you for bringing this to my attention. I am currently traveling but I will diagnose the issue as soon as possible. Did you fill out all of the information in the curl command?

**Reply by ChristopherOleson** [reply]
Yes I did.

**Reply by JEMcats** [reply]
I just edited the original message, there was one misplaced comma at the end of the JSON. The request seems to be working for me now.

**Reply by ChristopherOleson** [reply]
Thank you the request seems to be working but the output isn't returning any values. Do you have an example of expected values returned?

**Reply by JEMcats** [reply]
I probably should have put this in the original message, the timestamp section is just a placeholder you can use a [timestamp generator](https://timestampgenerator.com) to get a new timestamp, the first timestamp is the start date+time and the second one is the current date+time for the section of the cloud list to get.
